pkg/build/strategies/strategies.go

```diff
@@ -24,6 +25,18 @@ func Strategy(client docker.Client, config *api.Config, overrides build.Override
 
 	startTime := time.Now()
 
+	if len(config.WithBuilder) != 0 {
+		builder, err = external.New(config, fileSystem)
+		if err != nil {
+			buildInfo.FailureReason = utilstatus.NewFailureReason(
+				utilstatus.ReasonGenericS2IBuildFailed,
+				utilstatus.ReasonMessageGenericS2iBuildFailed,
+			)
+			return nil, buildInfo, err
+		}
+		return builder, buildInfo, nil
+	}
+
 	if len(config.AsDockerfile) != 0 {
```

Comment:
So, if I did a 
```
s2i build --as-dockerfile=path/to/write/dockerfile --with-builder=podman
```
first the control enters

```
if len(config.WithBuilder) != 0 {

}
```
and then the control again enters
```
if len(config.AsDockerfile) != 0 {

}
```

Reply:
In this case, we would like to short-circuit it first on `--with-builder`. By using this option we will generate the `Dockerfile` on behalf of the user, and thus `if len(config.WithBuilder) != 0 {` will return the builder.

Notice if option `--as-dockerfile` is informed without `--with-builder` it works like before (like in `master` branch).

Reply:
I understand. 
Would the dockerfile generation happen twice in this flow?

Reply:
Nope, since each of those `if`s are executing `return`.

Reply:
Ah right, I missed that! thank you.